capability.go

```diff
@@ -455,6 +461,10 @@ type ClientState struct {
 	Brand Brand
 	// IsPromise is true if the client has not resolved yet.
 	IsPromise bool
+	// Arbitrary metadata. Note that, if a Client is a promise,
+	// when it resolves its metadata will be replaced with that
+	// of its resolution.
```

Comment:
Note that I'm not 100% sure these semantics are actually what is desired, or at least it seems like we might want to still be able to access the un-resolved version of this. `State()` is supposed to return a snapshot, so perhaps we also want `OriginalState()` which would recover pre-resolution state (in which case if `OrigState().IsPromise == false` then `State() == OrigState()`)?

...I feel like I don't totally understand the broader use of this, and I kindof expect whatever decisions we make here we'll likely discover we need to tweak exactly how this works a bit before stabilizing.

Reply:
👍  Copy that.

Perhaps consider adding a `TODO` comment?

Reply:
Done.